### ISSUE TITLE
AttributeError: module 'torch.mps' has no attribute 'is_autocast_enabled'

### ISSUE THREAD

**Issue by tmm1** [pytorch/pytorch#104478]
### 🐛 Describe the bug

with pytorch nightly, seeing this error during MPS usage

```
  File "torch/utils/checkpoint.py", line 192, in forward
    ctx.device_autocast_kwargs, ctx.cpu_autocast_kwargs = _get_autocast_kwargs(
  File "torch/utils/checkpoint.py", line 170, in _get_autocast_kwargs
    "enabled": device_module.is_autocast_enabled(),

AttributeError: module 'torch.mps' has no attribute 'is_autocast_enabled'
```

seems like it started with https://github.com/pytorch/pytorch/pull/99626

### Versions

PyTorch version: 2.1.0.dev20230630
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.4 (arm64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: version 3.25.2
Libc version: N/A

Python version: 3.10.12 (main, Jun  7 2023, 00:38:32) [Clang 14.0.3 (clang-1403.0.22.14.1)] (64-bit runtime)
Python platform: macOS-13.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Ultra

Versions of relevant libraries:
[pip3] numpy==1.25.0
[pip3] torch==2.1.0.dev20230630
[pip3] torchaudio==2.1.0.dev20230630
[pip3] torchvision==0.16.0.dev20230630
[conda] numpy                     1.21.5                   pypi_0    pypi
[conda] numpydoc                  1.5.0           py310hca03da5_0  
[conda] torch                     2.1.0.dev20230601          pypi_0    pypi
[conda] torchvision               0.16.0.dev20230601          pypi_0    pypi

cc @ezyang @gchanan @zou3519 @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by tmm1** [comment]
another similar one:

```
  File "torch/autograd/function.py", line 506, in apply                                                                                                      
    return super().apply(*args, **kwargs)  # type: ignore[misc]                                                            
  File "torch/utils/checkpoint.py", line 203, in forward                                                                                                     
    if device_module._initialized:                                                                                         
AttributeError: module 'torch.mps' has no attribute '_initialized'
```

cc @heidongxianhua @soulitzer

**Comment by tmm1** [comment]
I think some of these issues will be addressed by https://github.com/pytorch/pytorch/pull/99272

**Comment by heidongxianhua** [comment]
yeah, I have checked the related code, there are two points we should do to support checkpointing for MPS:
1. the first one is to add autocast support for MPS;
2. and we should to add some utils function to torch.mps module, such as torch.mps.device, torch.mps.set_rng_state and ..

**Comment by heidongxianhua** [comment]
> I think some of these issues will be addressed by #99272

And as for mps to support AMP, maybe we could use the normalized  API https://github.com/pytorch/pytorch/pull/104495, not add API like `get_autocast_mps_dtype`?

**Comment by tmm1** [comment]
cc @birch-san I wonder if #104787 is enough to ditch [mps_autocast_stub](https://github.com/Birch-san/mpt-play/blob/main/src/mps_autocast_stub.py)
